salt/modules/file.py

```diff
@@ -1881,132 +1896,110 @@ def line(path, content=None, match=None, mode=None, location=None,
-                            out.append(content)
-                        elif line.find(after) > -1:
-                            raise CommandExecutionError('Found more than one line between '
-                                                        'boundaries "before" and "after".')
-                body = os.linesep.join(out)
+                idx = body.index(after[0])
+                if idx < (len(body) - 1) and body[idx + 1] == before[0]:
+                    body.insert(idx + 1, _set_line_indent(body[idx], _set_line_eol(body[idx], content), indent))
+                else:
+                    raise CommandExecutionError('Found more than one line between '
+                                                'boundaries "before" and "after".')
 
         elif before and not after:
-            _assert_occurrence(body, before, 'before')
-            body = body.split(os.linesep)
-            out = []
-            for idx in range(len(body)):
-                if body[idx].find(before) > -1:
-                    prev = (idx > 0 and idx or 1) - 1
-                    out.append(_get_line_indent(body[idx], content, indent))
-                    if _starts_till(out[prev], content) > -1:
-                        del out[prev]
-                out.append(body[idx])
-            body = os.linesep.join(out)
+            _assert_occurrence(before, 'before')
+
+            idx = body.index(before[0])
+            if body[idx-1].find(content) < 0:
+                body.insert(idx, _set_line_indent(body[idx], _set_line_eol(body[idx], content), indent))
 
         elif not before and after:
-            _assert_occurrence(body, after, 'after')
-            body = body.split(os.linesep)
-            skip = None
-            out = []
-            for idx in range(len(body)):
-                if skip != body[idx]:
-                    out.append(body[idx])
-
-                if body[idx].find(after) > -1:
-                    next_line = idx + 1 < len(body) and body[idx + 1] or None
-                    if next_line is not None and _starts_till(next_line, content) > -1:
-                        skip = next_line
-                    out.append(_get_line_indent(body[idx], content, indent))
-            body = os.linesep.join(out)
+            _assert_occurrence(after, 'after')
+
+            idx = body.index(after[0])
+            next_line = idx + 1 < len(body) and body[idx + 1] or None
+            if next_line is None or _starts_till(next_line, content) < 0:
+                body.insert(idx+1, _set_line_indent(body[idx], _set_line_eol(body[idx], content), indent))
 
         else:
             raise CommandExecutionError("Wrong conditions? "
                                         "Unable to ensure line without knowing "
                                         "where to put it before and/or after.")
 
-    changed = body_before != hashlib.sha256(salt.utils.stringutils.to_bytes(body)).hexdigest()
+    if body:
+        body.append(body.pop().rstrip('\r\n'))
```

Comment:
Would `os.linesep` be more appropriate here?

Reply:
I believe the idea is to rstrip either:

```
>>> 'fubar'.rstrip('ar')
'fub'
>>> 'fubar'.rstrip('ra')
'fub'
```

Reply:
It would not be more appropriate.
Line ending in this place is not OS dependant. It might be CRLF line end on machine with os.linesep = LF.

Reply:
OK.